File: showroom/src/figs/sine.ts
import type { Figure } from 'plotive'

export default function (): Figure {
    const x = Array.from({ length: 500 }, (_, i) => i / 499 * 2 * Math.PI);
    const y = x.map(x => Math.sin(x));

    return {
        title: "Sine Wave",
        plot: {
            series: {
                type: "line",
                x,
                y,
                name: "y=sin(x)",
            },
            xAxis: {
                title: "x",
                ticks: "pimultiple",
                grid: "auto",
            },
            yAxis: {
                title: "sin(x)",
                ticks: "auto",
                grid: "auto",
            },
            legend: "in-top-right",
        }
    }
}

File: showroom/src/figs/sine.rs
use std::f64::consts::PI;

use plotive::{data, des};

mod common;

fn main() {
    let fig = des::series::Line::new(des::data_src_ref("x"), des::data_src_ref("y"))
        .with_name("y=sin(x)")
        .into_plot()
        .with_x_axis(
            des::Axis::new()
                .with_title("x".into())
                .with_ticks(
                    des::axis::Ticks::new()
                        .with_locator(des::axis::ticks::PiMultipleLocator::default().into()),
                )
                .with_grid(Default::default()),
        )
        .with_y_axis(
            des::Axis::new()
                .with_title("y".into())
                .with_ticks(Default::default())
                .with_grid(Default::default()),
        )
        .with_legend(des::plot::LegendPos::InTopRight.into())
        .into_figure()
        .with_title("Sine wave".into());

    let x: Vec<f64> = (0..=360).map(|t| t as f64 * PI / 180.0).collect();
    let y = x.iter().map(|x| x.sin()).collect();

    let data_source = data::TableSource::new()
        .with_f64_column("x".into(), x)
        .with_f64_column("y".into(), y);

    common::process_figure(&fig, &data_source, None, "sine");
}
